azkaban-exec-server/src/test/java/azkaban/dag/TestDagProcessor.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.dag;
+
+import java.util.concurrent.CountDownLatch;
+
+public class TestDagProcessor implements DagProcessor {
+
+  private final StatusChangeRecorder statusChangeRecorder;
+  private final CountDownLatch dagFinishedLatch;
+
+
+  TestDagProcessor(final CountDownLatch dagFinishedLatch,
+      final StatusChangeRecorder statusChangeRecorder) {
+    this.dagFinishedLatch = dagFinishedLatch;
+    this.statusChangeRecorder = statusChangeRecorder;
+  }
+
+  @Override
+  public void changeStatus(final Dag dag, final Status status) {
+    System.out.println(dag);
```

Comment:
system out

Reply:
This is intentional. 
Any downside?

Reply:
Discussed with Cheng offline. 
He agreed that it is ok to use in test code.